code/controllers/subsystem/hijack.dm

```diff
@@ -0,0 +1,371 @@
+
+#define EVACUATION_TYPE_NONE 0
+#define EVACUATION_TYPE_ADDITIVE 1
+#define EVACUATION_TYPE_MULTIPLICATIVE 2
+
+#define HIJACK_ANNOUNCE "ARES Emergency Procedures"
+#define XENO_HIJACK_ANNOUNCE "You sense something unusual..."
+
+#define EVACUATION_STATUS_NOT_INITIATED 0
+#define EVACUATION_STATUS_INITIATED 1
+
+#define HIJACK_OBJECTIVES_NOT_STARTED 0
+#define HIJACK_OBJECTIVES_STARTED 1
+#define HIJACK_OBJECTIVES_COMPLETE 2
+
+SUBSYSTEM_DEF(hijack)
+	name   = "Hijack"
+	wait   = 2 SECONDS
+	flags  = SS_KEEP_TIMING
+	priority   = SS_PRIORITY_HIJACK
+
+	///Required progress to evacuate safely via lifeboats
+	var/required_progress = 100
+
+	///Current progress towards evacuating safely via lifeboats
+	var/current_progress = 0
+
+	/// How much progress is required to early launch
+	var/early_launch_required_progress = 25
+
+	///The estimated time left to get to the safe evacuation point
+	var/estimated_time_left = 0
+
+	///Areas that are marked as having progress, assoc list that is progress_area = boolean, the boolean indicating if it was progressing or not on the last fire()
+	var/list/progress_areas = list()
+
+	///The areas that need cycled through currently
+	var/list/current_run = list()
+
+	///The progress of the current run that needs to be added at the end of the current run
+	var/current_run_progress_additive = 0
+
+	///Holds what the current_run_progress_additive should be multiplied by at the end of the current run
+	var/current_run_progress_multiplicative = 1
+
+	///Holds the progress change from last run
+	var/last_run_progress_change = 0
+
+	///Holds the next % point progress should be announced, increments on itself
+	var/announce_checkpoint = 25
+
+	///What stage of evacuation we are currently on
+	var/evac_status = EVACUATION_STATUS_NOT_INITIATED
+
+	///What stage of hijack are we currently on
+	var/hijack_status = HIJACK_OBJECTIVES_NOT_STARTED
+
+	///Whether or not evacuation has been disabled by admins
+	var/evac_admin_denied = FALSE
+
+	/// If TRUE, self destruct has been unlocked and is possible with a hold of reactor
+	var/sd_unlocked = FALSE
+
+	/// Admin var to manually prevent self destruct from occurring
+	var/admin_sd_blocked = FALSE
+
+	/// Maximum amount of fusion generators that can be overloaded at once for a time benefit
+	var/maximum_overload_generators = 18
+
+	/// How many generators are currently overloaded
+	var/overloaded_generators = 0
+
+	/// How long the manual self destruct will take on the high end
+	var/sd_max_time = 15 MINUTES
+
+	/// How long the manual self destruct will take on the low end
+	var/sd_min_time = 5 MINUTES
+
+	/// How much time left until SD detonates
+	var/sd_time_remaining = 0
+
+	/// If the self destruct has/is detonating
+	var/sd_detonated = FALSE
+
+/datum/controller/subsystem/hijack/Initialize()
+	. = ..()
+	RegisterSignal(SSdcs, COMSIG_GLOB_GENERATOR_SET_OVERLOADING, PROC_REF(on_generator_overload))
```

Comment:
`## WARNING: Hijack subsystem does not implement Initialize() or it returns ..(). If the former is true, the SS_NO_INIT flag should be set for this subsystem.`

```suggestion
/datum/controller/subsystem/hijack/Initialize(timeofday)
	RegisterSignal(SSdcs, COMSIG_GLOB_GENERATOR_SET_OVERLOADING, PROC_REF(on_generator_overload))
	return SS_INIT_SUCCESS
```